packages/init/src/tasks/checkPackageManager.ts

```diff
@@ -0,0 +1,24 @@
+import { findUpSync } from 'find-up';
+
+import logger from '../utils/logger.js';
+
+function isProjectUsingYarn(cwd: string) {
+  return findUpSync('yarn.lock', { cwd });
+}
+
+/**
+ * Determines which package manager to use
+ *
+ * @param cwd current working directory
+ * @returns package manager name
+ */
+export default function checkPackageManager(cwd: string) {
+  const yarnLockPath = isProjectUsingYarn(cwd);
+
+  if (yarnLockPath) {
+    logger.info('Using yarn as package manager');
+    return 'yarn';
+  }
+  logger.info('Using npm as package manager');
+  return 'npm';
```

Comment:
PNPM is another possibility here, we can add an additional check for it:

```ts
findUpSync('pnpm-lock.yaml', {cwd})
```

Or we can use an existing package to detect package manager (first hit from google, have a lot of downloads but last release was almost 2 years ago): https://www.npmjs.com/package/detect-package-manager

Reply:
I've kind of omitted `pnpm` as it doesn't work well in React-Native ecosystem (yet).

I'll check out the `detect-package-manager` and see whether it makes sense here. 
I've resorted to using method of detecting package manager almost identical to that of RN CLI.

Reply:
In theory if someone is using pnpm and it works for his usecase, I see no reason to not incldue it as one of the options and utilize that. It's needed only for the step of installing dependencies anyways